### ISSUE TITLE
`wing simulate` command and `cloud.Output` resource

### ISSUE THREAD

**Issue by Chriscbr** [winglang/wing#2430]
### Feature Spec

`cloud.Output` is a preflight class allows you to export structured data about your app. For terraform this compiles to a [Terraform output value](https://developer.hashicorp.com/terraform/cdktf/concepts/variables-and-outputs#output-values), and for AWS CDK this compiles to a [CloudFormation output value](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_core.CfnOutput.html).

Example:

```js
bring cloud;

let api = new cloud.Api();
new cloud.Output(api.url) as "MyEndpoint";
```

When compile this to terraform and apply the terraform configuration, you'll see the output listed:

```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

MyEndpoint = "http://some-url.amazonaws.com/"
```

### Use Cases

Sharing/reference values between multiple (Terraform/Cloudformation/etc.) apps.

Exposing information about your app to the command line.

### Component

SDK

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by ekeren** [comment]
@Chriscbr should this be in the cloud namespace or an std namespace ?

**Comment by skyrpex** [comment]
The console could also have a dedicated place for these outputs.

**Comment by eladb** [comment]
Let's get this implemented.

@ainvoner @skyrpex 

I think these outputs should be included in the preview environment comment.

For example, this could be the way we show the endpoints an app is exposing.

**Comment by eladb** [comment]
@staycoolcall911 can we prioritize

**Comment by staycoolcall911** [comment]
@eladb yes of course. Updated to p1. We should soon find some time to go over all our p1s

**Comment by skorfmann** [comment]
what's the naming gonna be? Personally I like `Output`, `Endpoint` implies some functionality aspect to me.

**Comment by skorfmann** [comment]
> `Output` is more general and may have more requirements.

Could you elaborate on the potential requirements which popped up during your discussions?

**Comment by staycoolcall911** [comment]
It wasn't a long discussion. It's just the general idea that a Wing program's `Output` is broader than just an endpoint. It could be a tarball or any other artifact, for example.
BTW, the name endpoint is also not great and is ambiguous with `cloud.Api` endpoints, so maybe we can find a better name for this which is not `Output` or `Endpoint`.

**Comment by eladb** [comment]
Let's go with `cloud.Endpoint` for now. I think there's value in higher level modeling. We can always add `cloud.Output` later as a more general purpose thing.

Our use case right now is to model the idea of an **endpoint** and explicitly expose these endpoints in Wing Cloud.

So let's go with `cloud.Endpoint`.

The API is straightforward:

```js
new cloud.Endpoint(url);
```

No?

**Comment by eladb** [comment]
Renamed title to `cloud.Endpoint`

**Comment by eladb** [comment]
Let's keep this issue to track `cloud.Output` and split `cloud.Endpoint` into a separate issue.

An endpoint is more than just an output. Our use case is to be able to show endpoints in the console, expose them from preview environments and from local dev machines, maybe bind them to domains, etc.

So they need to be explicitly modeled.

I am not against also introducing cloud.Output (now or later) and define an output for all endpoints, but it's not required for the endpoint use case.

**Comment by staycoolcall911** [comment]
@eladcon I'm taking you off the issue because I know it's not on your plans anymore. This issue is up for grabs
